networking-calico/networking_calico/plugins/ml2/drivers/calico/test/test_monitor_thread.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2025 Tigera, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+networking_calico.plugins.ml2.drivers.calico.test.test_monitor_thread
+~~~~~~~~~~~
+
+Unit tests for the thread that monitors the periodic resync thread.
+"""
+from datetime import datetime, timedelta
+import mock
+import unittest
+
+import networking_calico.plugins.ml2.drivers.calico.test.lib as lib
+from networking_calico.plugins.ml2.drivers.calico import mech_calico
+
+
+INITIAL_EPOCH = 0
+TEST_MAX_INTERVAL = 30
+
+
+class TestResyncMonitorThread(lib.Lib, unittest.TestCase):
+    """Tests for the driver's resync monitor thread logic."""
+
+    def setUp(self):
+        super(TestResyncMonitorThread, self).setUp()
+
+        # thread logic mocks
+        self.driver.elector = mock.Mock()
+        self.sleep_patcher = mock.patch("eventlet.sleep")
+        self.mock_sleep = self.sleep_patcher.start()
+
+        # log mocks
+        self.log_error = mock.patch.object(mech_calico.LOG, "error").start()
+        self.log_info = mock.patch.object(mech_calico.LOG, "info").start()
+        self.log_debug = mock.patch.object(mech_calico.LOG, "debug").start()
+
+        # resync mocks
+        self.driver.subnet_syncer = mock.Mock()
+        self.driver.policy_syncer = mock.Mock()
+        self.driver.endpoint_syncer = mock.Mock()
+        self.driver.provide_felix_config = mock.Mock()
+        self.compaction_patcher = mock.patch(
+            "networking_calico.plugins.ml2.drivers"
+            ".calico.mech_calico.check_request_etcd_compaction"
+        )
+
+    def tearDown(self):
+        self.sleep_patcher.stop()
+        super(TestResyncMonitorThread, self).tearDown()
+
+    def increment_epoch(self, curr_epoch):
+        return lambda _: setattr(self.driver, "_epoch", curr_epoch + 1)
+
+    def test_monitor_does_nothing_when_not_master(self):
+        """Test that a driver that is not master does not monitor."""
+        self.driver.elector.master.return_value = False
+        self.mock_sleep.side_effect = self.increment_epoch(INITIAL_EPOCH)
```

Comment:
I'm not following this.  Is it important to set the epoch to INITIAL_EPOCH+1 ?

Reply:
Oh, I guess it's a way to avoid ending up with multiple resync and monitor threads running?